#26643 Implement command execution ordering for push and pull

jgambarios opened the pull request:
The push and pull commands have been modified to allow subcommand execution in a custom order. A new method, `getOrder`, has been added in both `DotPush` and `DotPull` interfaces and implemented in all subclasses. Corresponding unit tests have also been updated to ensure that commands with lower `getOrder` values are executed first.

jgambarios committed:
#26643 Implement command execution ordering for push and pull

The push and pull commands have been modified to allow subcommand execution in a custom order. A new method, 'getOrder', has been added in both DotPush and DotPull interfaces and implemented in all subclasses. Corresponding unit tests have also been updated to ensure that commands with lower 'getOrder' values are executed first.

fabrizzio-dotCMS commented on tools/dotcms-cli/cli/src/main/java/com/dotcms/cli/command/files/FilesPull.java:
```diff
@@ -155,4 +155,9 @@ public Optional<String> getCustomMixinName() {
 
+    @Override
+    public int getOrder() {
+        return 3;
```
Wouldn't it be better if we make this into an enum
public enum ExecOrder {
    SITE(1), CONTENT_TYPE(2).. 
}
and then simply return order?

fabrizzio-dotCMS reviewed (commented):
I made a suggestion lets see what others think

jgambarios committed:
#26643 Implement enum for command execution order

Introduced an `ApplyCommandOrder` enum to determine the execution order of subcommands in push and pull operations. This replaces the hardcoded order integers in the overridden `getOrder` method across various classes, improving consistency and maintainability.
